src/libraries/System.Private.CoreLib/src/System/IO/BufferedStream.cs

```diff
@@ -82,34 +89,47 @@ public BufferedStream(Stream stream, int bufferSize)
                 throw new ObjectDisposedException(null, SR.ObjectDisposed_StreamClosed);
         }
 
+        internal BufferedStream(Stream stream, int bufferSize, bool actLikeFileStream) : this(stream, bufferSize)
+        {
+            _actLikeFileStream = actLikeFileStream;
```

Comment:
All the special cases where we check for `_actLikeFileStream` in this file, seem to be complicating the code and I worry that it could end up becoming difficult to maintain in the future, just like it happened to `FileStream`. So I wonder if we should invest in also applying the Strategy pattern to `BufferedStream` (not on this PR, but on another one in the near future).

Reply:
I totally agree that it complicates the `BufferedStream` source code. I just don't have a better idea of how we could do this without loosing perf